signac/contrib/errors.py

```diff
@@ -31,3 +31,8 @@ def __init__(self, job_ids):
 class StatepointParsingError(Error, RuntimeError):
     "Indicates an error that occurred while tyring to identify a state point."
     pass
+
+
+class IncompatibleSchemaVersion(Error):
+    "The project's schema version is incompatible with this version."
```

Comment:
If we use the word "version," we should always specify what kind of version (schema/signac package) to avoid confusion.
```suggestion
    "The project's schema version is incompatible with this version of signac."
```